source/agora/test/TimeBlockInterval.d

```diff
@@ -20,48 +20,28 @@ import agora.consensus.data.Transaction;
 import agora.crypto.Hash;
 import agora.test.Base;
 
+import std.range;
+import std.algorithm;
+
 ///
 unittest
 {
-    TestConf conf = { txs_to_nominate : 2, block_interval_sec : 10 };
+    TestConf conf = { block_interval_sec : 10 };
     auto network = makeTestNetwork!TestAPIManager(conf);
     network.start();
     scope(exit) network.shutdown();
     scope(failure) network.printLogs();
     network.waitForDiscovery();
 
-    auto nodes = network.clients;
-    auto node_1 = nodes[0];
-
-    auto spendable = network.blocks[0].txs
-        .filter!(tx => tx.isPayment)
-        .map!(tx => iota(tx.outputs.length)
-            .map!(idx => TxBuilder(tx, cast(uint)idx)))
-        .joiner().take(8).array;
-
-    auto txs = spendable
-        .map!(txb => txb.refund(WK.Keys.Genesis.address).sign())
-        .array;
-
-    // 8 transactions is enough for 4 blocks with 2 txs each
-    txs.each!(tx => nodes[0].putTransaction(tx));
-
-    // check for propagation
-    nodes.each!(node =>
-       txs.each!(tx =>
-           node.hasAcceptedTxHash(hashFull(tx)).retryFor(4.seconds)
-    ));
-
-    // time updated, block height 1
-    const b0 = nodes[0].getBlocksFrom(0, 2)[0];
-
-    void checkHeight(Height height)
-    {
-        network.waitForPreimages(b0.header.enrollments, height, 30.seconds);
-        network.setTimeFor(height);
-        network.assertSameBlocks(height);
-    }
-
-    // Check for adding blocks 1 to 4
-    [1, 2, 3, 4].each!(h => checkHeight(Height(h)));
+    auto node_1 = network.clients.front;
+
+    auto startTime = node_1.getNetworkTime();
+    // create 8 txs enough for 4 blocks with 2 txs each
+    genesisSpendable.map!(txb =>
+        txb.sign()).array
+        .chunks(2).enumerate.each!((en) {
```

Comment:
```suggestion
    genesisSpendable.map!(txb => txb.sign())
        .chunks(2).enumerate.each!((en) {
```

The array call is not necessary, is it ?

Reply:
I access the elements by index later. I could use `drop(height -1).take(1)` but seemed clearer this way.

Reply:
Checking.

Reply:
Yes not needed. Removing.